lib/core/src/Cardano/Wallet/Primitive/Model.hs

```diff
@@ -376,7 +378,9 @@ prefilterBlock b u0 = runState $ do
         , direction = dir
         , slotNo = b ^. #header . #slotNo
         , blockHeight = b ^. #header . #blockHeight
-        , amount = Quantity amt
+        -- fixme: ADP-347
+        -- fixme: why on earth do we have both Coin and Quantity "lovelace" Natural?
```

Comment:
At the very beginning, the `Quantity` type was only used in the Api as a way to generate pretty JSON. It has one day escaped its cage and spread over the rest of the codebase as a quick replacement for newtypes over numeric types.. So in some places we have `Coin`, in some `Word64`, in some `Quantity "lovelace" Natural` or `Quantity "lovelace" Word64` or also `Fee` which is very much the same thing as `Coin`.